README.md

```diff
@@ -150,6 +150,45 @@ class User < ActiveRecord::Base
 end
 ```
 
+Several callbacks are provided in order to hook in your application-specific
+logic for mapping the auth data to your User model. These methods will be
+executed whenever a user logs in.
+
+In order to set simple user attributes based on auth data, override the
+`attributes_from_auth` method. Call `super` when you do so in order to pick
+up the minimum set of required user attributes:
+
+```ruby
+class User < ActiveRecord::Base
+  devise :g5_authenticatable
+
+  def attributes_from_auth(auth_data)
+    super(auth_data).merge({
+      name: "#{auth_data.info.first_name} #{auth_data.info.last_name}"
+    })
+  end
+end
+```
+
+If you want to customize the logic for mapping auth role data (which is
+ignored by default), override the `update_roles_from_auth` method:
+
+```ruby
+class User < ActiveRecord::Base
+  devise :g5_authenticatable
+
+  def update_roles_from_auth(auth_data)
+    auth_data.extra.roles.each do |r|
```

Comment:
auth_data.extra.roles has the possibility to be a Huge hash of keys when a customer has a lot of explicit roles for a lot of locations 
This was an issue in our one of our applications when we were using cookies instead of table sessions.
Keep in mind 
https://www.pivotaltracker.com/story/show/129389397